hazelcast/src/hazelcast/client/proxy.cpp

```diff
@@ -39,6 +39,8 @@ const std::chrono::milliseconds imap::UNSET{ -1 };
 reliable_topic::reliable_topic(const std::string& instance_name,
                                spi::ClientContext* context)
   : proxy::ProxyImpl(reliable_topic::SERVICE_NAME, instance_name, context)
+  , execution_service_(
+      context->get_client_execution_service().shared_from_this())  
   , executor_(context->get_client_execution_service().get_user_executor())
```

Comment:
`execution_service_` is usable at the construction of `executor_` so I suggest to change it as :

```suggestion
  , executor_(execution_service->get_user_executor())
```
****
It is not crucial but it makes less verbose.

Reply:
Fixed